core/iwasm/include/gc_export.h

```diff
@@ -0,0 +1,512 @@
+    uint32 global_index;
+    uint32 ref_index;
+    int64 i64;
+    uint64 u64;
+    float32 f32;
+    float64 f64;
+    V128 v128;
+    wasm_obj_t *gc_obj;
+    uint32 type_index;
+    struct {
+        uint32 type_index;
+        uint32 N;
+    } array_new_canon_fixed;
+} WASMValue;
+#endif /* end of WASM_VALUE_DEFINED */
+
+typedef union WASMValue wasm_value_t;
+
+/* Reference type, the layout is same as WasmRefType in wasm.h */
+typedef struct wasm_ref_type_t {
+    wasm_value_type_t value_type;
+    bool nullable;
+    int32_t heap_type;
+} wasm_ref_type_t;
+
+/**
+ * Local object reference that can be traced when GC occurs. All
+ * native functions that need to hold WASM objects which may not be
+ * referenced from other elements of GC root set may be hold with
+ * this type of variable so that they can be traced when GC occurs.
+ * Before using such a variable, it must be pushed onto the stack
+ * (implemented as a chain) of such variables, and before leaving the
+ * frame of the variables, they must be poped from the stack.
+ */
+typedef struct WASMLocalObjectRef {
+    /* Previous local object reference variable on the stack */
+    struct WASMLocalObjectRef *prev;
+    /* The reference of WASM object hold by this variable */
+    wasm_obj_t val;
+} WASMLocalObjectRef, wasm_local_obj_ref_t;
+
+struct WASMType;
+struct WASMFuncType;
+struct WASMStructType;
+struct WASMArrayType;
+
+typedef struct WASMType *wasm_defined_type_t;
+typedef struct WASMFuncType *wasm_func_type_t;
+typedef struct WASMStructType *wasm_struct_type_t;
+typedef struct WASMArrayType *wasm_array_type_t;
+
+struct WASMExternrefObject;
+struct WASMAnyrefObject;
+struct WASMStructObject;
+struct WASMArrayObject;
+struct WASMFuncObject;
+
+typedef struct WASMExternrefObject *wasm_externref_obj_t;
+typedef struct WASMAnyrefObject *wasm_anyref_obj_t;
+typedef struct WASMStructObject *wasm_struct_obj_t;
+typedef struct WASMArrayObject *wasm_array_obj_t;
+typedef struct WASMFuncObject *wasm_func_obj_t;
+typedef uintptr_t wasm_i31_obj_t;
+
+/* Defined type related operations */
+
+/**
+ * Get the defined type count of a WASM module
+ */
+WASM_RUNTIME_API_EXTERN uint32_t
+wasm_get_defined_type_count(const wasm_module_t module);
+
+/**
+ * Get the specified defined type of a WASM module
+ */
+WASM_RUNTIME_API_EXTERN wasm_defined_type_t
+wasm_get_defined_type(const wasm_module_t module, uint32_t index);
+
+/**
+ * Check whether a defined type is a function type
+ */
+WASM_RUNTIME_API_EXTERN bool
+wasm_defined_type_is_func_type(const wasm_defined_type_t def_type1);
+
+/**
+ * Check whether a defined type is a struct type
+ */
+WASM_RUNTIME_API_EXTERN bool
+wasm_defined_type_is_struct_type(const wasm_defined_type_t def_type1);
+
+/**
+ * Check whether a defined type is an array type
+ */
+WASM_RUNTIME_API_EXTERN bool
+wasm_defined_type_is_array_type(const wasm_defined_type_t def_type1);
+
+/**
+ * Get parameter count of a function type
+ */
+WASM_RUNTIME_API_EXTERN uint32
```

Comment:
In `wasm_export.h`, we use `uint32_t` rather than `uint32`, should we keep the same rule in this file?

Reply:
yes, had better use `xxx_t` types